kategory-core/src/main/kotlin/kategory/typeclasses/Order.kt

```diff
@@ -1,32 +1,110 @@
 package kategory
 
+/**
+ * The [Order] type class is used to define a total ordering on some type [F] and is defined by being able to fully determine order between two instances.
+ *
+ * [Order] is a subtype of [Eq] and defines [eqv] in terms of [compare].
+ *
+ * @see [Eq]
+ * @see <a href="http://kategory.io/docs/typeclasses/order/">Order documentation</a>
+ */
 interface Order<F> : Eq<F>, Typeclass {
 
     /**
      * Compare [a] with [b]. Returns an Int whose sign is:
      * - negative if `x < y`
      * - zero     if `x = y`
      * - positive if `x > y`
+     *
+     * @param a object to compare with [b]
+     * @param b object to compare with [a]
+     * @returns zero objects are equal, a negative number if [a] is less than [b], or a positive number if [a] greater than [b].
      */
     fun compare(a: F, b: F): Int
 
+    /** @see [Eq.eqv] */
     override fun eqv(a: F, b: F): Boolean = compare(a, b) == 0
 
+    /**
+     * Check if [a] is `lower than` [b]
+     *
+     * @param a object to compare with [b]
+     * @param b object to compare with [a]
+     * @returns true if [a] is `lower than` [b] and false otherwise
+     */
     fun lt(a: F, b: F): Boolean = compare(a, b) < 0
```

Comment:
should we add operator extension syntax for all these ops? `>`, `<` ....? https://kotlinlang.org/docs/reference/operator-overloading.html#operator-overloading

Reply:
Operator overloading is only allowed for functions with 1 parameter. So currently there is no way to add syntax since we have to pass in `Order` as a parameter.